dev-libs/boost/boost-1.88.0.ebuild

```diff
@@ -0,0 +1,459 @@
+		"predef"
+		# AttributeError: property '<unnamed Boost.Python function>' of 'X' object has no setter
+		"python"
+		# vec_access.hpp:95:223: error: static assertion failed: Boost QVM static assertion failure
+		"qvm"
+		# regex_timer.cpp:19: ../../../boost/timer.hpp:21:3: error: #error This header is
+		#   deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress
+		#   this error.)
+		"regex"
+		# in function `boost::archive::tmpnam(char*)': test_array.cpp:(.text+0x108):
+		#   undefined reference to `boost::filesystem::detail::unique_path(...)'
+		"serialization"
+		# TuTestMain.cpp(22) fatal error: in "test_main_caller( argc_ argv )":
+		#   std::runtime_error: Event was not consumed!
+		"statechart"
+		# erase_tests.cpp:(.text+0x44cce): undefined reference to
+		#   tbb::detail::r1::execution_slot(tbb::detail::d1::execution_data const*)
+		"unordered"
+		# t_5_007.cpp(22): error: could not find include file: boost/version.hpp
+		"wave"
+	)
+
+	if use mpi; then
+		ewarn "Enabling graph_parallel tests due to USE=mpi"
+		libs_excluded=("${libs_excluded[@]/graph_parallel}")
+	fi
+
+	if use test-full; then
+		ewarn "Enabling expensive tests - this will take a long time!"
+		libs_excluded=("${libs_excluded[@]/geometry}")
+	fi
+
+	# libraries which must be patched with an additonal main()
+	local libs_needpatch=(
+		"accumulators"
+	)
+
+	# prepare to find libraries without failing candidates
+	local findlibs="find ${BUILD_DIR}/libs -maxdepth 1 -mindepth 1 -type d"
+	for excl in ${libs_excluded[@]}; do
+	   findlibs+=" -not -name ${excl} "
+	done
+
+	# collect libraries to test in sorted order
+	local libs=($(${findlibs} | sort))
+
+	echo; echo -n "Running the following tests: "
+	for lib in ${libs[@]}; do echo -n "$(basename ${lib}) "; done
+	echo
+
+	echo; echo -n "Skipping: "
+	for lib in ${libs_excluded[@]}; do echo -n "${lib} "; done
+	echo
+
+	# Some test suites have no main because normally boost.test can
+	# automatically initialize & run them. Since this only works for
+	# statically linked builds/tests we use an explicit list of tests
+	# which need patching with an additional main().
+	# Determining this dynamically is not really possible.
+	for lib in "${libs_needpatch[@]}"; do
+		# move into library test dir
+		pushd "${BUILD_DIR}/libs/${lib}/test" >/dev/null || die
+			# find all test cases and patch them
+			testcases=($(find . -name "*.cpp"))
+			for testcase in "${testcases[@]}"; do
+				# add main() to bootstrap old-style test suite
+				cat "${FILESDIR}/unit-test-main.cpp" >> ${testcase}
+			done
+		popd >/dev/null
+	done
+
+	# finally build & run all test suites
+	for lib in "${libs[@]}"; do
+		# skip libraries without test directory
+		[[ ! -d "${lib}/test" ]] && continue
+
+		# move into library test dir & run all tests
+		pushd "${lib}/test" >/dev/null || die
+			echo; echo "Running tests in: "$(pwd)
+			ejam --prefix="${EPREFIX}"/usr "${TEST_OPTIONS[@]}" || die
```

Comment:
Maybe `edo` this so you can drop the die.

Reply:
Yes, that's actually much better since it also swallows more compilation output, which is just distracting.